test/Sema/availability_versions.swift

```diff
@@ -985,6 +985,19 @@ func useUnavailableExtension() {
       // expected-note@-1 {{add 'if #available' version check}}
 }
 
+// Availability of synthesized designated initializers.
+
+@available(OSX, introduced: 10.51)
+class WidelyAvailableBase {
+  init() {}
+
+  @available(OSX, introduced: 10.52)
+  init(thing: ()) {}
+}
+
+@available(OSX, introduced: 10.53)
+class EsotericSmallBatchHipsterThing : WidelyAvailableBase {}
```

Comment:
Thanks for fixing this! Can you add a test in the opposite direction, where the initializer is 10.53 and the subclass is 10.52? (I know that would break a lot of other things too, but we should have a test for it anyway.)